src/main/java/org/zalando/nakadi/service/subscription/SubscriptionStreamerFactory.java

```diff
@@ -56,17 +65,23 @@ public SubscriptionStreamerFactory(
         this.metricRegistry = metricRegistry;
         this.zkClientFactory = zkClientFactory;
         this.eventStreamWriterProvider = eventStreamWriterProvider;
+        this.eventTypeRepository = eventTypeRepository;
+        this.authorizationService = authorizationService;
     }
 
     public SubscriptionStreamer build(
             final String subscriptionId,
             final StreamParameters streamParameters,
             final SubscriptionOutput output,
             final AtomicBoolean connectionReady,
-            final BlacklistService blacklistService) throws NoSuchSubscriptionException, ServiceUnavailableException,
+            final BlacklistService blacklistService,
+            final Client client) throws NoSuchSubscriptionException, ServiceUnavailableException,
             InternalNakadiException, NoSuchEventTypeException {
 
         final Subscription subscription = subscriptionDbRepository.getSubscription(subscriptionId);
+
+        authorizeSubscriptionRead(eventTypeRepository, authorizationService, client, subscription);
```

Comment:
The permission check is inside build method, which is to build the object, not to check the permission. It would be nice to have it in service layer, but we do not have it exactly here. We can move it to controller level, but there we need subscription. Maybe we can move retrieval of subscription to controller level or create service level for that?